pkg/machines/components/diskAdd.jsx

```diff
@@ -397,6 +397,10 @@ export class AddDiskModalBody extends React.Component {
     onAddClicked() {
         const { vm, dispatch, provider, close, vms, storagePools } = this.props;
 
+        // validate
+        if (!this.state.storagePoolName)
```

Comment:
This should actually never be non-set.
If it does it means we have some bug somewhere in the contructor of the DiskAdd component, where the initially selected storage pool should be set.

Reply:
This actually shows when user does not have any storage pools on their host. Then instead of dropdown we show text: _No Storage Pools available_. In that case, state.storagePoolName is undefined, however we do not prevent user from clicking "Add" button and calling attachDisk action, and it results in some error.
This should prevent from attachDisk action being called.